mtgjson5/set_builder.py

```diff
@@ -44,10 +44,26 @@
     UuidCacheProvider,
     WhatsInStandardProvider,
 )
+from .providers.enrichment_provider import EnrichmentProvider
 from .utils import get_str_or_none, load_local_set_data, url_keygen
 
 LOGGER = logging.getLogger(__name__)
 
+# Singleton instance cache for EnrichmentProvider
+_enrichment_provider_instance = None
```

Comment:
No globals please, let it be passed around as needed (Which the @singleton will help with)